.upstream-tests/test/cuda/pipeline_memcpy_async_thread_scope_generic.h

```diff
@@ -107,19 +107,23 @@ __host__ __device__ __noinline__
 void test_select_destination()
 {
     test_select_pipeline<Scope, T, SourceSelector, local_memory_selector>();
-#ifdef __CUDA_ARCH__
-    test_select_pipeline<Scope, T, SourceSelector, shared_memory_selector>();
-    test_select_pipeline<Scope, T, SourceSelector, global_memory_selector>();
-#endif
+    NV_DISPATCH_TARGET(
+        NV_IS_DEVICE, (
+            test_select_pipeline<Scope, T, SourceSelector, shared_memory_selector>();
+            test_select_pipeline<Scope, T, SourceSelector, global_memory_selector>();
+        )
+    )
 }
 
 template <cuda::thread_scope Scope, class T>
 __host__ __device__ __noinline__
 void test_select_source()
 {
     test_select_destination<Scope, T, local_memory_selector>();
-#ifdef __CUDA_ARCH__
-    test_select_destination<Scope, T, shared_memory_selector>();
-    test_select_destination<Scope, T, global_memory_selector>();
-#endif
+    NV_DISPATCH_TARGET(
+        NV_IS_DEVICE, (
+            // test_select_destination<Scope, T, shared_memory_selector>();
+            // test_select_destination<Scope, T, global_memory_selector>();
```

Comment:
Is this a leftover from something?

Reply:
Yeah, I'm on the lookout for more of these. This was from some earlier testing on NVC++.